templates/getting_started/faq.md

```diff
@@ -493,51 +493,40 @@ model.fit(dataset, epochs=10, validation_data=val_dataset)
 
 ---
 
-### How can I regularly save Keras models during training?
+### How can I ensure my training with Keras can withstand program interruptions?
 
 To ensure the ability to recover from an interrupted training run at any time (fault tolerance),
-you should use a callback that regularly saves your model to disk. You should also set up
-your code to optionally reload that model at startup. Here's a simple example.
+you should use a `tf.keras.callbacks.experimental.BackupAndRestore` that regularly saves your training progress,
+including the epoch number and weights, to disk, and loads it the next time you call `Model.fit`.
 
 ```python
-import os
+import tensorflow as tf
 from tensorflow import keras
 
-# Prepare a directory to store all the checkpoints.
-checkpoint_dir = './ckpt'
-if not os.path.exists(checkpoint_dir):
-    os.makedirs(checkpoint_dir)
-
-
-def make_model():
-    # Create a new linear regression model.
-    model = keras.Sequential([keras.layers.Dense(1)])
-    model.compile(optimizer='adam', loss='mse')
-    return model
-
-
-def make_or_restore_model():
-    # Either restore the latest model, or create a fresh one
-    # if there is no checkpoint available.
-    checkpoints = [checkpoint_dir + '/' + name
-                   for name in os.listdir(checkpoint_dir)]
-    if checkpoints:
-        latest_checkpoint = max(checkpoints, key=os.path.getctime)
-        print('Restoring from', latest_checkpoint)
-        return keras.models.load_model(latest_checkpoint)
-    print('Creating a new model')
-    return make_model()
-
-
-model = make_or_restore_model()
-callbacks = [
-    # This callback saves a SavedModel every 100 batches.
-    # We include the training loss in the folder name.
-    keras.callbacks.ModelCheckpoint(
-        filepath=checkpoint_dir + '/ckpt-loss={loss:.2f}',
-        save_freq=100)
-]
-model.fit(train_data, epochs=10, callbacks=callbacks)
+class InterruptingCallback(keras.callbacks.Callback):
+  """A callback to intentionally introduce interruption to training."""
+  def on_epoch_end(self, epoch, log=None):
+    if epoch == 15:
+      raise RuntimeError('Interruption')
+
+model = keras.Sequential([keras.layers.Dense(10)])
+optimizer = keras.optimizers.SGD()
+model.compile(optimizer, loss="mse")
+
+x = tf.random.uniform((24, 10))
+y = tf.random.uniform((24,))
+dataset = tf.data.Dataset.from_tensor_slices((x, y)).repeat().batch(2)
+
+backup_callback = keras.callbacks.experimental.BackupAndRestore(
+    backup_dir='/tmp/backup')
+try:
+  model.fit(dataset, epochs=20, steps_per_epoch=5, 
+            callbacks=[backup_callback, InterruptingCallback()])
+except RuntimeError:
+  print('***Handling interruption***')
+  # This continues at the epoch where it left off.
```

Comment:
This isn't entirely true due to the epoch counter, no?

Reply:
So this is what would happen if we execute this code:

```
Epoch 1/20
5/5 [==============================] - 1s 4ms/step - loss: 0.7054
Epoch 2/20
5/5 [==============================] - 0s 5ms/step - loss: 0.4781
Epoch 3/20
5/5 [==============================] - 0s 5ms/step - loss: 0.5767
Epoch 4/20
5/5 [==============================] - 0s 3ms/step - loss: 0.4884
Epoch 5/20
5/5 [==============================] - 0s 3ms/step - loss: 0.4787
Epoch 6/20
5/5 [==============================] - 0s 5ms/step - loss: 0.4814
Epoch 7/20
5/5 [==============================] - 0s 5ms/step - loss: 0.3595
Epoch 8/20
5/5 [==============================] - 0s 2ms/step - loss: 0.4798
Epoch 9/20
5/5 [==============================] - 0s 7ms/step - loss: 0.3371
Epoch 10/20
5/5 [==============================] - 0s 3ms/step - loss: 0.4095
Epoch 11/20
5/5 [==============================] - 0s 2ms/step - loss: 0.3282
Epoch 12/20
5/5 [==============================] - 0s 3ms/step - loss: 0.3167
Epoch 13/20
5/5 [==============================] - 0s 3ms/step - loss: 0.4042
Epoch 14/20
5/5 [==============================] - 0s 2ms/step - loss: 0.2524
Epoch 15/20
5/5 [==============================] - 0s 2ms/step - loss: 0.3485
Epoch 16/20
5/5 [==============================] - 0s 2ms/step - loss: 0.2691
***Handling interruption***
Epoch 17/20
5/5 [==============================] - 0s 3ms/step - loss: 0.3449
Epoch 18/20
5/5 [==============================] - 0s 2ms/step - loss: 0.2131
Epoch 19/20
5/5 [==============================] - 0s 3ms/step - loss: 0.3047
Epoch 20/20
5/5 [==============================] - 0s 2ms/step - loss: 0.2291
```

It picked up the next epoch because we interrupted at epoch end. If the interruption happens in the middle of epoch N, it would restart at epoch N.